test/y2partitioner/dialogs/partition_resize_test.rb

```diff
@@ -157,18 +165,98 @@ def find_label(contents, text)
       context "when max size is selected" do
         let(:current_widget) { max_size_widget }
 
-        it "updates the partition with the max size" do
-          subject.store
-          expect(partition.size).to eq(max_size)
+        RSpec.shared_examples "max sizes" do
+          context "and the max size causes an end-aligned partition" do
+            let(:max_size) { aligned_max_size }
+
+            it "updates the partition with the max size" do
+              subject.store
+              expect(partition.size).to eq max_size
+              expect(partition.end_aligned?).to eq(true)
+            end
+          end
+
+          context "and the max size causes a not end-aligned partition" do
+            let(:max_size) { not_aligned_max_size }
+
+            it "updates the partition with the max aligned size" do
+              subject.store
+              expect(partition.size).to eq adjusted_size
+              expect(partition.end_aligned?).to eq(true)
+              expect(adjusted_size).to be <= max_size
```

Comment:
NP: I know it is the same, but I would replace the last `expect` by `expect(partition.size).to be <= max_size` (but it LGTM at it is now)